CONTRIBUTING.md

```diff
@@ -7,27 +7,80 @@ We'd love for you to contribute to this repository. Before you start, we'd like
 * Before you submit your pull request, search the repository for an open or closed Pull Request that relates to your submission. You don't want to duplicate effort. 
 * For commiting in your Pull Request, You can checkout [Commits](#commits) for more.
 
-### Coding Rules
+# Coding Rules
 
-#### Swift style
+## Swift style
 * Swift code should generally follow the conventions listed at https://github.com/raywenderlich/swift-style-guide.
   * Exception: we use 4-space indentation instead of 2.
   * This is a loose standard. We do our best to follow this style
 
-#### Whitespace
+## Whitespace
 * New code should not contain any trailing whitespace.
 * We recommend enabling both the "Automatically trim trailing whitespace" and "Including whitespace-only lines" preferences in Xcode (under Text Editing).
 * <code>git rebase --whitespace=fix</code> can also be used to remove whitespace from your commits before issuing a pull request.
 
-### Commits
+## Commits
 * Each commit should have a single clear purpose. If a commit contains multiple unrelated changes, those changes should be split into separate commits.
 * If a commit requires another commit to build properly, those commits should be squashed.
 * Follow-up commits for any review comments should be squashed. Do not include "Fixed PR comments", merge commits, or other "temporary" commits in pull requests.
+
+
+### Commit strategy
+
+We prefer to keep out commit history linear (meaning avoiding noisy merge
+commits). To keep your branch up to date, follow these steps:
+
+```bash
+# while on your PR branch
+git checkout develop
+git pull --rebase
+git checkout my-pr-branch
+git rebase develop
+git push origin my-pr-branch --force[-with-lease]
+```
+
+##Before Pushing
+### Running Danger Locally
+
+You can run danger locally to see what it would output by running the following in the root of the repository.
+
+```bash
+swift run danger-swift pr [some-pr-url]
+```
+
+### UI Tests (Mozillans/Pocketeers Only)
+
+UI Tests are split into 3 test plans. As you built UI tests you should ensure that your test is enabled in 1 of the UI test plans to be run on CI in such a way as to balance the 3 test plans.
+
+
+### Localization
+
+Pocket localization is handled by Smartling. The project is currently setup in [Single Branch Mode](https://help.smartling.com/hc/en-us/articles/360008152513-GitHub-Connector-Overview#SingleBranchModeTranslationFlow) against the `develop` branch. Everytime a commit is made to `develop` Smartling will analyze the branch and determine if it needs to start a translation job. If it does, it will begin automatically and make a PR back against the repo with the needed translations.
+
+#### Adding Strings
+
+Pocket uses [swiftgen](https://github.com/SwiftGen/SwiftGen#strings) to generate a [Strings.swift](./PocketKit/Sources/PocketKit/Strings.swift) file from our English [Localizable.strings](./PocketKit/Sources/PocketKit/Resouces/en.lproj/Localizable.strings) file.
```

Comment:
on the localization note (separate from this PR though), should we discuss if we should make it available across the project as opposed to just PocketKit?

Reply:
We should just make Localization its own package.